var/spack/repos/builtin/packages/hdf-eos2/package.py

```diff
@@ -0,0 +1,99 @@
+    # The data in version_list is used to generate versions and urls
+    # In basename expansions, 0 is raw version,
+    # 1 is for version with dots => underscores
+    version_list = [
+        {'version': '2.20v1.00',
+            'sha256': 'cb0f900d2732ab01e51284d6c9e90d0e852d61bba9bce3b43af0430ab5414903',
+            'basename': 'HDF-EOS{0}.tar.Z'},
+        {'version': '2.19b',
+            'sha256': 'a69993508dbf5fa6120bac3c906ab26f1ad277348dfc2c891305023cfdf5dc9d',
+            'basename': 'hdfeos{1}.zip'}
+    ]
+
+    for vrec in version_list:
+        ver = vrec['version']
+        sha256 = vrec['sha256']
+        version(ver, sha256=sha256)
+
+    variant('shared', default=True,
+            description='Build shared libraries (can be used with +static)')
+    variant('static', default=True,
+            description='Build static libraries (can be used with +shared)')
+
+    conflicts('~static', when='~shared',
+              msg='At least one of +static or +shared must be set')
+
+    # Build dependencies
+    depends_on('hdf')
+
+    # The standard Makefile.am, etc. add a --single_module flag to LDFLAGS
+    # to pass to the linker.
+    # That appears to be only recognized by the Darwin linker, remove it
+    # if we are not running on darwin/
+    if sys.platform != "darwin":
+        patch('hdf-eos2.nondarwin-no-single_module.patch')
+
+    def url_for_version(self, version):
+        vrec = [x for x in self.version_list
+                if x['version'] == version.dotted.string]
+        if vrec:
+            fname = vrec[0]['basename'].format(version.dotted,
+                                               version.underscored)
+            sha256 = vrec[0]['sha256']
+            myurl = self.url.format(sha256, fname)
+            return myurl
+        else:
+            sys.exit('ERROR: cannot generate URL for version {0};'
+                     'version/checksum not found in version_list'.format(
+                         version))
+
+    def configure_args(self):
+        extra_args = []
+
+        # Package really wants h4cc to be used
+        extra_args.append('CC={0}/bin/h4cc -Df2cFortran'.format(
+            self.spec['hdf'].prefix))
+
+        # We always build PIC code
+        extra_args.append('--with-pic')
+
+        # Set shared/static appropriately
+        extra_args.extend(self.enable_or_disable('shared'))
+        extra_args.extend(self.enable_or_disable('static'))
+
+        # Provide config args for dependencies
+        extra_args.append('--with-hdf4={0}'.format(self.spec['hdf'].prefix))
+        if self.spec['jpeg']:
+            extra_args.append('--with-jpeg={0}'.format(
+                self.spec['jpeg'].prefix))
+        if self.spec['zlib']:
+            extra_args.append('--with-zlib={0}'.format(
+                self.spec['zlib'].prefix))
+        if self.spec['szip']:
+            extra_args.append('--with-szlib={0}'.format(
+                self.spec['szip'].prefix))
```

Comment:
If the package directly links to these libraries, you should add explicit dependencies on them.

Reply:
I am not clear on what you mean by "directly links".  I confess I am not an expert on the hdf-eos packages (I am a sysadmin building for an user), but the build instructions are not listing these as direct dependencies of hdf-eos2 (and similar for hdf-eos5) but instead are dependencies for the underlying hdf (hdf5 for hdf-eos5) dependency.  

Since the hdf package builds static libraries only, the build of hdf-eos2 requires the proper flags to link to the dependencies of hdf.  I therefore considered jpeg, zlib, and szip to be "indirect" dependencies and so did not list them as formal spack dependencies.  I.e., if hdf does not depend on them, neither will hdf-eos2.

Currently, hdf has an optional dependency (via the szip variant) on szip, and always depends on jpeg and zlib.  I wrapped the setting of the configuration/link flags in conditionals so that the flags are only set if the spack spec shows that there is a dependency on the relevant package, presumably because hdf depends on it.  I.e., if hdf depends on zlib, jpeg, or szip, the proper link flags are set, and if not, the flags are not set.

I will defer to your judgement and make jpeg, zlib, and (depending on hdf variant) szip proper spack dependencies if you still think it best.

Reply:
Alright, let's go with what you have for now and if it becomes problematic we can always change it in the future.